### DISCUSSION TITLE
merging / supporting / complementing obsidian.nvim

### DISCUSSION THREAD

**Discussion by anandkumar89** [Feel-ix-343/markdown-oxide#40] [General]
obsidian.nvim is already trying address same concern. efforts could be better utilised maintaining single system of great plugin. One less plugin to choose from and less overhead to deal for maintainers. Isn't it win-win for all!

I really like being able to manage obsidian vault from terminal. Currently I use obsidian.nvim to manage vault : link files, move around links etc. In addition, Zotlit provides ways to integrate zotero : citing, importing notes etc. It works although not perfect.

**Comment by Feel-ix-343** [comment]
Hey thanks for the comment!

I have actually been in contact with the author of Obsidian.nvim, in which we have discussed making markdown oxide a backend for it - to stop using rg completions. 

The next step is to write up an RFC on their discussions page, and I just haven't gotten to this yet. Im happy to see there is some want for it, so I will probably get it up soon.

For this merge, what are you thinking specifically?

**Reply by anandkumar89** [reply]
Not much clear on how to share goals. much of obsidian features fits right in realm of LSP (reference autocompletions, refactoring). handling references/linking would address - linking, backlinks and forward links and refactoring would encompass changing file names, headlines, tags uniformly across vault. These are almost all the features obsidian.nvim currently provides but could be considered within LSP definition. 

other things obsidian.nvim handles/could handle are 
- handling daily notes ✔️ 
- templates and snippets ✔️ 
- extracting notes ✔️ 
- adding images (already supported via pngpaste) ✔️ 
- syntax-highlighting (identifying math zones for snippet support - latex suite in obsidian),
    - one step further could provide math referencing support by labels (figure labels, equation labels) (while obsidian uses mathjax, but limitation of code mirror restricts them from using numbered equations supported by mathjax)
    - identifying code blocks (too much to expect?)
- little fancy to expect but callouts are widely used in obsidian community could be supported
- searching across vault, switchers. ✔️ 
- Other obsidian features (epwalsh maintains that obsidian.nvim is not a replacement for obsidian, but ...) 
    - audio recording with timestamped notes 
    - bookmarking files 
    - version control using git


these developments could be fast tracked if two are merged (complement each other) with separate goals to address.

**Reply by Feel-ix-343** [reply]
Thanks for this! Many of these obsidian features would likely be offloaded to the server - causing markdown oxide to be an advanced backend for obsidian.nvim

There is also a ton of potential in the realm of completions, which can be supported by a more powerful and custom parser that a server had access to. 
- completions for callouts (like you said; this is quite easy and I will implement it soon - possibly today)
- completions for dataview and obsidian query language
- completions for obsidian properties/metadata


And further, there are distant ideas to implement semantic search (through a vector database) and to rank link completions by contextual similarity. There are also features only available in a real backend (rust). 

Your comments are of great use, and I especially thank you for getting the ball rolling on this. I will open the RFC on the obsidian.nvim repo by the end of tomorrow - the author is expecting it.

**Reply by Feel-ix-343** [reply]
note that as of c06e778 moxide supports callouts

**Reply by anandkumar89** [reply]
still awaiting packaged release on Mason otherwise home-brew is forcing me to update python when try to install rust. I'll just hold my breath for few more days till it's released there.

Can't wait to test the features!! excited :)

**Reply by Feel-ix-343** [reply]
The PR is up and we are just waiting. [Here](https://github.com/mason-org/mason-registry/pull/5175) it is FYI

**Reply by anandkumar89** [reply]
Installed via Mason (guess packaged version is not accepted yet). 

Does it seem to work out of the box? I've obsidian.nvim installed and I open any file in obsidian from anywhere (whatever be the cwd) and in most cases cwd is not the obsidian vault. 

I see that nowhere obsidian vault path is being asked, so somehow it is being inferred, but it doesn't seem to work for me, probably issue with setup. Not sure how I can work with obsidian vault without cwd being anywhere in obsidians vault / root folder.

**Reply by Feel-ix-343** [reply]
Yes you must be in your obsidian vault path right now. I'll add a config features for this and adjust the nvim-lspconfig root_dir in a PR

**Reply by anandkumar89** [reply]
With binary released on Mason, I was able to test out the plugin. Initially I disabled obsidian.nvim as there are clash of features.

Yet to test all the features, works nicely. Great work!

I assume root dir feature isn't implemented yet and I need to be in obsidian cwd each time I want to take a note. I don't use obsidian alone for standalone notes store, I usually am writing a report and obsidian is where I take/put chunks of new information. 

my workflow : Using neovim mostly to type reports in latex (thanks to Gilles Castell's). I open obsidian doc (obsidian.nvim provides telescope pickers) 
- open daily note and add few notes / todos
- search for notes I've taken earlier. 
During all these, I'm still in a different directory. I almost never need to cd to different directory.

**Reply by Feel-ix-343** [reply]
Hey the lspconfig root directory detection was just updated. See if it works better for you now. It should detect a .git, .obsidian, or .moxide.toml file as the root directory

**Reply by anandkumar89** [reply]
I've been using this with obsidian.nvim and so far it's working great. Only I need some way to stop obsidian.nvim's completion.

**Reply by Feel-ix-343** [reply]
(reply is below)

**Comment by Feel-ix-343** [comment]
Also there was been some discussion about bibliography support here #12

**Reply by anandkumar89** [reply]
this is interesting! using bibtex files is reliable and robust idea.

designing code so that it further can be extended to accommodate for more functionalities would be good. e.g. [zotcite](https://github.com/jalvesaq/zotcite) automatically identifies bib files or zotero directory to complete citations. uses zotero keys in the citations text  to 
- open corresponding pdfs with keybindings.
- extract highlighted notes in pdfs
- extract notes from zotero database

<img width="723" alt="Screenshot 2024-03-30 at 1 37 11 AM" src="https://github.com/Feel-ix-343/markdown-oxide/assets/12571250/493beb1a-bb0b-48d3-a207-3ec843eb3edb">

**Reply by Feel-ix-343** [reply]
I like these

Some of them would only be possible in the form of a neovim plugin, however -- showing the importance of the merge!

**Reply by Idobenhamo** [reply]
Something like this can be amazing for the bib files integration

**Comment by Feel-ix-343** [comment]
I know you can disable nvim cmp in the config!

**Reply by anandkumar89** [reply]
Thanks, now I've disabled it. Works great!
